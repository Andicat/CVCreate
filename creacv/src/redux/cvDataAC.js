const CV_BLOCK_ADD = 'CV_BLOCK_ADD';
const CV_BLOCK_DELETE = 'CV_BLOCK_DELETE';
const CV_BLOCK_MOVE = 'CV_BLOCK_MOVE';
const CV_BLOCK_ACTIVATE = 'CV_BLOCK_ACTIVATE';

const cvBlock_add = function(block) {
  return {
    type: CV_BLOCK_ADD,
    block:block,
  };
}

const cvBlock_delete = function(block) {
  return {
    type: CV_BLOCK_ADD,
    block:block,
  };
}

const cvBlock_move = function(block) {
  return {
    type: CV_BLOCK_MOVE,
    block:block,
  };
}

const cvBlock_activate = function(block) {
  return {
    type: CV_BLOCK_ACTIVATE,
    block:block,
  };
}

export {
  cvBlock_add, CV_BLOCK_ADD,
  cvBlock_delete, CV_BLOCK_DELETE,
  cvBlock_move, CV_BLOCK_MOVE,
  cvBlock_activate, CV_BLOCK_ACTIVATE,
}
