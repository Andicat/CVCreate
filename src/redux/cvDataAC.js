const CV_BLOCK_ADD = 'CV_BLOCK_ADD';
const CV_BLOCK_DELETE = 'CV_BLOCK_DELETE';
const CV_BLOCK_MOVE = 'CV_BLOCK_MOVE';
const CV_BLOCK_RESIZE = 'CV_BLOCK_RESIZE';
const CV_BLOCK_ACTIVATE = 'CV_BLOCK_ACTIVATE';
const CV_BLOCK_ACTIVATE_MULTI = 'CV_BLOCK_ACTIVATE_MULTI';
const CV_BLOCK_COPY = 'CV_BLOCK_COPY';
const CV_BLOCK_SET_SIZE = 'CV_BLOCK_SET_SIZE';
const CV_ELEMENT_ACTIVATE = 'CV_ELEMENT_ACTIVATE';
const CV_STYLE_UPDATE = 'CV_STYLE_UPDATE';
const CV_TEXT_UPDATE = 'CV_TEXT_UPDATE';
const CV_BLOCK_SEND_BACK = 'CV_BLOCK_SEND_BACK';
const CV_BLOCKS_ALIGN_TOP = 'CV_BLOCKS_ALIGN_TOP';
const CV_BLOCKS_ALIGN_BOTTOM = 'CV_BLOCKS_ALIGN_BOTTOM';
const CV_BLOCKS_ALIGN_LEFT = 'CV_BLOCKS_ALIGN_LEFT';
const CV_BLOCKS_ALIGN_RIGHT = 'CV_BLOCKS_ALIGN_RIGHT';
const CV_BLOCKS_ALIGN_VERTICAL = 'CV_BLOCKS_ALIGN_VERTICAL';
const CV_BLOCKS_ALIGN_HORISONTAL = 'CV_BLOCKS_ALIGN_HORISONTAL';
const CV_BLOCKS_DISTRIBUTE_VERTICAL = 'CV_BLOCKS_DISTRIBUTE_VERTICAL';
const CV_BLOCKS_DISTRIBUTE_HORISONTAL = 'CV_BLOCKS_DISTRIBUTE_HORISONTAL';
const CV_BLOCKS_ALIGN_WIDTH = 'CV_BLOCKS_ALIGN_WIDTH';
const CV_BLOCKS_ALIGN_HEIGHT = 'CV_BLOCKS_ALIGN_HEIGHT';
const CV_BLOCKS_GROUP = 'CV_BLOCKS_GROUP';
const CV_BLOCK_UNGROUP = 'CV_BLOCK_UNGROUP';
const CV_BLOCK_LOCK = 'CV_BLOCK_LOCK';
const CV_LOAD = 'CV_LOAD';
const CV_INIT = 'CV_INIT';
const CV_SET_USER = 'CV_SET_USER';
const CV_SET_LINK = 'CV_SET_LINK';
const CV_BLOCK_LINK = 'CV_BLOCK_LINK';
const TEMPLATE_ADD = 'TEMPLATE_ADD';

const cvBlock_add = function(block) {
    return {
        type: CV_BLOCK_ADD,
        block:block,
    };
}

const cvBlock_delete = function(blockId) {
    return {
        type: CV_BLOCK_DELETE,
        blockId:blockId,
    };
}

const cvBlock_move = function(blockId,shiftTop,shiftLeft) {
    return {
        type: CV_BLOCK_MOVE,
        blockId:blockId,
        shiftTop: shiftTop,
        shiftLeft: shiftLeft,
    };
}

const cvBlock_resize = function(blockId,shiftHeight,shiftWidth) {
    return {
        type: CV_BLOCK_RESIZE,
        blockId:blockId,
        shiftHeight: shiftHeight,
        shiftWidth: shiftWidth,
    };
}

const cvBlock_activate = function(blockId,target) {
    return {
        type: CV_BLOCK_ACTIVATE,
        blockId:blockId,
        target:target,
    };
}

const cvElement_activate = function(style,elementId) {
    return {
        type: CV_ELEMENT_ACTIVATE,
        elementId:elementId,
        style:style,
    };
}

const cvStyle_update = function(blockId,styleName,styleValue) {
    return {
        type: CV_STYLE_UPDATE,
        blockId: blockId,
        styleName: styleName,
        styleValue: styleValue,
    };
}

const cvElement_textUpdate = function(blockId,textValue) {
    return {
        type: CV_TEXT_UPDATE,
        blockId: blockId,
        textValue: textValue,
    };
}

const cvBlock_sendBack = function(blockId) {
    return {
        type: CV_BLOCK_SEND_BACK,
        blockId: blockId,
    };
}

const cvBlock_copy = function(blockId) {
    return {
        type: CV_BLOCK_COPY,
        blockId: blockId,
    };
}

const cvBlock_setSize = function(blockId,height,width) {
    return {
        type: CV_BLOCK_SET_SIZE,
        blockId: blockId,
        width:width,
        height:height,
    };
}

const cvBlock_activateMulti = function(blockId) {
    return {
        type: CV_BLOCK_ACTIVATE_MULTI,
        blockId: blockId,
    };
}

const cvBlocks_align = function(mode) {
    switch (mode) {
        case 'top':
            return { type: CV_BLOCKS_ALIGN_TOP };
        case 'bottom':
            return { type: CV_BLOCKS_ALIGN_BOTTOM };
        case 'left':
            return { type: CV_BLOCKS_ALIGN_LEFT };
        case 'right':
            return { type: CV_BLOCKS_ALIGN_RIGHT };
        case 'vertical':
            return { type: CV_BLOCKS_ALIGN_VERTICAL };
        case 'horisontal':
            return { type: CV_BLOCKS_ALIGN_HORISONTAL };
        default:
    }
}

const cvBlocks_distribute = function(mode) {
    switch (mode) {
        case 'vertical':
            return { type: CV_BLOCKS_DISTRIBUTE_VERTICAL };
        case 'horisontal':
            return { type: CV_BLOCKS_DISTRIBUTE_HORISONTAL };
        default:
    }
}

const cvBlocks_alignSize = function(mode) {
    switch (mode) {
        case 'width':
            return { type: CV_BLOCKS_ALIGN_WIDTH };
        case 'height':
            return { type: CV_BLOCKS_ALIGN_HEIGHT };
        default:
    }
}

const cvBlocks_group = function() {
    return { type: CV_BLOCKS_GROUP };
}

const cvBlock_ungroup = function(blockId) {
    return { 
        type: CV_BLOCK_UNGROUP,
        blockId: blockId,
    };
}

const cvBlock_lock = function(blockId) {
    return { 
        type: CV_BLOCK_LOCK,
        blockId: blockId,
    };
}

const cv_load = function(blocks,style) {
    return { 
        type: CV_LOAD,
        blocks:blocks,
        style:style,
    };
}

const cv_init = function(blocks,style,user,link,templatesData,templatesUser) {
    return { 
        type: CV_INIT,
        blocks:blocks,
        style:style,
        user:user,
        link:link,
        templatesData:templatesData,
        templatesUser:templatesUser,
    };
}

const cv_setUser = function(user) {
    return { 
        type: CV_SET_USER,
        user:user,
    };
}

const cv_setLink = function(link) {
    return { 
        type: CV_SET_LINK,
        link:link,
    };
}

const cvBlock_setLink = function(blockId,linkValue) {
    return { 
        type: CV_BLOCK_LINK,
        blockId:blockId,
        linkValue:linkValue,
    };
}

const templates_add = function(blockId) {
    return {
        type: TEMPLATE_ADD,
        blockId: blockId,
    };
}

export {
    cvBlock_add, CV_BLOCK_ADD,
    cvBlock_delete, CV_BLOCK_DELETE,
    cvBlock_move, CV_BLOCK_MOVE,
    cvBlock_resize, CV_BLOCK_RESIZE,
    cvBlock_activate, CV_BLOCK_ACTIVATE,
    cvBlock_activateMulti, CV_BLOCK_ACTIVATE_MULTI,
    cvBlock_sendBack, CV_BLOCK_SEND_BACK,
    cvBlock_copy, CV_BLOCK_COPY,
    cvBlock_setSize, CV_BLOCK_SET_SIZE,
    cvBlock_lock, CV_BLOCK_LOCK,
    cvBlock_setLink, CV_BLOCK_LINK,
    cvBlocks_align, CV_BLOCKS_ALIGN_TOP, CV_BLOCKS_ALIGN_BOTTOM, CV_BLOCKS_ALIGN_LEFT, CV_BLOCKS_ALIGN_RIGHT, CV_BLOCKS_ALIGN_VERTICAL, CV_BLOCKS_ALIGN_HORISONTAL,
    cvBlocks_distribute, CV_BLOCKS_DISTRIBUTE_VERTICAL, CV_BLOCKS_DISTRIBUTE_HORISONTAL,
    cvBlocks_alignSize, CV_BLOCKS_ALIGN_WIDTH, CV_BLOCKS_ALIGN_HEIGHT,
    cvBlocks_group, CV_BLOCKS_GROUP, 
    cvBlock_ungroup, CV_BLOCK_UNGROUP,
    cvElement_activate, CV_ELEMENT_ACTIVATE,
    cvStyle_update, CV_STYLE_UPDATE,
    cvElement_textUpdate, CV_TEXT_UPDATE,
    cv_load, CV_LOAD,
    cv_init, CV_INIT, 
    cv_setUser, CV_SET_USER,
    cv_setLink, CV_SET_LINK,
    templates_add, TEMPLATE_ADD
}
