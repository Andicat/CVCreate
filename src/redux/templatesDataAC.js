const TEMPLATE_LOAD = 'TEMPLATE_LOAD';

const templates_load = function(data) {
    return {
        type: TEMPLATE_LOAD,
        data: data,
    };
}

export { templates_load, TEMPLATE_LOAD }
