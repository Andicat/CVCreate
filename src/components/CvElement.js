import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {cvElement_activate, cvElement_textUpdate} from '../redux/cvDataAC';
import {createStyle, decodeStyle} from '../modules/utils';
import sprite from '../img/sprite.svg';

class CvElement extends React.PureComponent {

    static propTypes = {
        editable: PropTypes.bool,
        id: PropTypes.string.isRequired,
        blockId: PropTypes.number.isRequired,
        data: PropTypes.object,
        activeElementId: PropTypes.string,
        templateImageUrl: PropTypes.string,
    };

    onClick = () => {
        if (this.props.editable) {
            this.props.dispatch(cvElement_activate(this.props.data.style,this.props.id));
        }
    }

    onBlur = (evt) => {
        let textCurr = evt.target.innerText;
        if (this.props.data.text!==textCurr) {
            this.props.dispatch(cvElement_textUpdate(this.props.blockId,textCurr));
        }
    }

    render () {
        //console.log('render cv element',this.props.id);
        let isActive = (this.props.activeElementId===this.props.id);
        let style = createStyle(this.props.data.style);
        let positionTop;
        let positionLeft;
        let position;
        let styleText;
        if (this.props.data.link) {
            positionTop = style.top?style.top:(this.props.data.positionTop + 'px');
            positionLeft = style.left?style.left:(this.props.data.positionLeft + 'px');
            position = style.position;
            delete style.top;
            delete style.left;
            delete style.position;
        }
        let type = this.props.data.type;
        let className = 'cv__element cv__element--' + type + (isActive?' cv__element--active':'') + (this.props.data.direction?(' cv__element--' + type + '-' + this.props.data.direction):'');
        let elementCode = null;
        let decodedStyle;
        switch (type) {
            case 'image':
                decodedStyle = {};
                for (let s in this.props.data.style) {
                    decodedStyle[decodeStyle(s)] = this.props.data.style[s];
                }
                let src = (decodedStyle['file']) || this.props.templateImageUrl;
                elementCode = <img className={className} src={src} style={style} alt='' data-elem={true} onClick={this.onClick}/>;
                break;
            case 'text':
                let text = this.props.data.text;
                styleText = {color: style.color, fontFamily: style.fontFamily,
                    fontSize: style.fontSize, fontStyle:style.fontStyle,
                    fontWeight: style.fontWeight, padding: style.padding,
                    textAlign: style.textAlign, textDecoration:style.textDecoration,
                    textTransform: style.textTransform};
                if (this.props.data.list) {
                    elementCode = <ul style={{fontSize:style.fontSize, paddingLeft:style.fontSize}}>
                                    <li className={className} style={{color:style.colorlist}}>
                                        <span style={styleText} suppressContentEditableWarning={this.props.editable} contentEditable={this.props.editable} data-elem={true} onClick={this.onClick} onBlur={this.onBlur}>
                                            {text}
                                        </span>
                                    </li>
                                </ul>;
                } else {
                    elementCode = <span className={className} style={style} suppressContentEditableWarning={this.props.editable} contentEditable={this.props.editable} data-elem={true} onClick={this.onClick} onBlur={this.onBlur}>
                                    {text}
                                </span>;
                }
                break;
            case 'figure':
                elementCode = <div className={className} style={style} data-elem={true} onClick={this.onClick}></div>;;
                break;
            case 'progress':
                decodedStyle = {};
                for (let s in this.props.data.style) {
                    decodedStyle[decodeStyle(s)] = this.props.data.style[s];
                }
                let styleProgressBar = {backgroundColor: decodedStyle.maincolor, width:decodedStyle.progress+'%'};
                let progressBarCode = <div style={styleProgressBar}></div>;
                let styleProgressBg = {backgroundColor: decodedStyle.addcolor, width:(100-decodedStyle.progress)+'%'};
                let progressBgCode = <div style={styleProgressBg}></div>;
                elementCode = <div className={className} style={style} data-elem={true} onClick={this.onClick}>
                                {progressBarCode}
                                {progressBgCode}
                              </div>;
                break;
            case 'dots-row':
                decodedStyle = {};
                for (let s in this.props.data.style) {
                    decodedStyle[decodeStyle(s)] = this.props.data.style[s];
                }
                let styleMainDots = {backgroundColor: decodedStyle.maincolor, width:decodedStyle.radius + 'px', height:decodedStyle.radius + 'px', margin:'0 ' + decodedStyle.radius*0.5 + 'px', borderRadius: '50%'};
                let mainDotsCode = Array.from({length: decodedStyle.maincount}, (v,i) => <div key={i} style={styleMainDots}></div>);
                let styleAddDots = {...styleMainDots, backgroundColor: decodedStyle.addcolor};
                let addDotsCode = Array.from({length: decodedStyle.addcount}, (v,i) => <div key={i} style={styleAddDots}></div>);
                elementCode = <div className={className} style={style} data-elem={true} onClick={this.onClick}>
                                {mainDotsCode}
                                {addDotsCode}
                              </div>;
                break;
            case 'icon':
                    decodedStyle = {};
                    for (let s in this.props.data.style) {
                        decodedStyle[decodeStyle(s)] = this.props.data.style[s];
                    }
                    let styleElem = {position: style.position, width:style.width, height:style.height, top:style.top, left:style.left};
                    let styleIcon = {fill: decodedStyle.fill, width:decodedStyle.size +'px', height:decodedStyle.size +'px'};
                    styleText = {color: style.color, fontFamily: style.fontFamily,
                                    fontSize: style.fontSize, fontStyle:style.fontStyle,
                                    fontWeight: style.fontWeight, padding: style.padding,
                                    textAlign: style.textAlign, textDecoration:style.textDecoration,
                                    textTransform: style.textTransform};
                    let textIcon = this.props.data.text;
                    if (textIcon instanceof Array) {
                        textIcon = textIcon.map( (w,i) => <span key={i}>{w}{i<textIcon.length-1&&<br/>}</span>);
                    }
                    elementCode = <div className={className} style={styleElem} data-elem={true} onClick={this.onClick}>
                                    <svg style={styleIcon}>
                                        <use href={sprite + '#' + this.props.data.svg}></use>
                                    </svg>
                                    <span style={styleText} suppressContentEditableWarning={this.props.editable} contentEditable={this.props.editable} data-elem={true} onClick={this.onClick} onBlur={this.onBlur}>
                                        {textIcon}
                                    </span>
                                  </div>;
                    break;
            case 'group':
                let CvGroupElement = connect(mapStateToProps)(CvElement);
                elementCode = <div className={className} style={style}>
                                {this.props.data.elements.map( (e,i) => (
                                    <CvGroupElement key={'' + (e.id?e.id:i)} id={'' + (e.id?e.id:i)} blockId={this.props.blockId} editable={this.props.editable} data={e} activeElementId={this.props.activeElementId}></CvGroupElement>
                                ))}
                              </div>;
                break;
            case 'link':
                elementCode = <a className={className} style={style} href={this.props.data.href} suppressContentEditableWarning={this.props.editable} contentEditable={this.props.editable} data-elem={true} onClick={this.onClick} onBlur={this.onBlur}>
                                {this.props.data.text}
                                </a>;
                break;
            default:
                elementCode = null;
        }
        if (this.props.data.link) {
            let styleLink = {position:position, top:positionTop, left:positionLeft, width:this.props.data.width + 'px', height:this.props.data.height + 'px'};
            return (
                <a className='cv__element--link' href={this.props.data.link} style={styleLink}>
                    {elementCode} 
                </a>
            );
        }
        
        return elementCode;        
    }
}

const mapStateToProps = function (state) {
    return {
        templateImageUrl: state.cvData.templateImageUrl,
    };
};

export default connect(mapStateToProps)(CvElement);
