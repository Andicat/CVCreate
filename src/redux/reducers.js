import { combineReducers } from 'redux';

import cvDataReducer from "./cvDataReducer";

let combinedReducer=combineReducers({
    cvData: cvDataReducer,
});

export default combinedReducer;
