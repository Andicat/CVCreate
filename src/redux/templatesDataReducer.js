import { TEMPLATE_LOAD } from './templatesDataAC';

const initState = {
    templatesArr: [],
    imageUrl: null,
}

function templatesDataReducer(state = initState, action) {

    switch (action.type) {
        //load
        case TEMPLATE_LOAD: {
            let newState = {templatesArr:action.data.templates, imageUrl:action.data.image};
            return newState;
        }
        default:
            return state;
    }
}

export default templatesDataReducer;
