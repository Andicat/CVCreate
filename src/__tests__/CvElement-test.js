"use strict";

import {renderComponentWithStore, storeEmpty} from './utils-test';
import {cvElement_activate, cvElement_textUpdate} from '../redux/cvDataAC';

import CvElement from '../components/CvElement';

let component;

let propEditable = {editable:false};
let propId = {id:'1'};
let propBlockId = {blockId:1};
let propActiveElementId = {activeElementId:null};
let textStyleDefault = {font:'Roboto', color:'#000000', fontsize:'16', bold:false, italic:false, center:false, uppercase:false, underline:false, padding:{left:1,right:0,top:0,bottom:0}};

let propDataEmpty = {data: {}};
let propDataImage = {data: {type:'image', style:{file:'', opacity:1, borderRadius:'50%', bordercolor: '#E05B49', borderwidth: '3', borderStyle: 'solid'}}};
let propDataText = {data: {type:'text', text:'Text with background', style:{bgcolor:'#8e9fa0',...textStyleDefault, fontsize:'14'}}};
let propDataTextBlock = {data: {type:'group', elements:[
                                {type:'text', text:'Your position', style:{...textStyleDefault, fontsize:'18', bold:true}},
                                {type:'text', text:'Company', style:{...textStyleDefault, fontsize:'18'}},
                                {type:'text', text:'period', style:{...textStyleDefault,italic:true}},
                                {type:'text', text:'your competencies and results', style:{...textStyleDefault}}]}};
let propDataColorBlock = {data: {type:'figure', style:{bgcolor:'#6AABB5', opacity:1, borderRadius:'50%'}}};
let propDataDots = {data: {type:'dots-row', style:{maincolor:'#E05B49', addcolor:'#E6E6E6', radius:10, maincount:5, addcount: 3}}};
let propDataProgress = {data: {type:'progress', style:{maincolor:'#E05B49', addcolor:'#E6E6E6', progress:50}}};

let props;

describe('CvElement render', () => {

    it('Рендер без данных', () => {
        props = {...propEditable, ...propId, ...propBlockId, ...propDataEmpty, ...propActiveElementId};
        component = renderComponentWithStore(CvElement,storeEmpty,props);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Рендер картинки', () => {
        props = {...propEditable, ...propId, ...propBlockId, ...propDataImage, ...propActiveElementId};
        component = renderComponentWithStore(CvElement,storeEmpty,props);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Рендер текст', () => {
        props = {...propEditable, ...propId, ...propBlockId, ...propDataText, ...propActiveElementId};
        component = renderComponentWithStore(CvElement,storeEmpty,props);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Рендер текстового блока', () => {
        props = {...propEditable, ...propId, ...propBlockId, ...propDataTextBlock, ...propActiveElementId};
        component = renderComponentWithStore(CvElement,storeEmpty,props);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Рендер цветного блока', () => {
        props = {...propEditable, ...propId, ...propBlockId, ...propDataColorBlock, ...propActiveElementId};
        component = renderComponentWithStore(CvElement,storeEmpty,props);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Рендер блока с кружочками', () => {
        props = {...propEditable, ...propId, ...propBlockId, ...propDataDots, ...propActiveElementId};
        component = renderComponentWithStore(CvElement,storeEmpty,props);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Рендер блока прогресса', () => {
        props = {...propEditable, ...propId, ...propBlockId, ...propDataProgress, ...propActiveElementId};
        component = renderComponentWithStore(CvElement,storeEmpty,props);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Рендер текста с возможностью правки', () => {
        props = {editable:true, ...propId, ...propBlockId, ...propDataText, ...propActiveElementId};
        component = renderComponentWithStore(CvElement,storeEmpty,props);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Рендер активного элемента', () => {
        props = {...propEditable, ...propId, ...propBlockId, ...propDataText, activeElementId:'1'};
        component = renderComponentWithStore(CvElement,storeEmpty,props);
        expect(component.toJSON()).toMatchSnapshot();
    });
});

describe('CvElement actions', () => {

    it('Activate', () => {
        storeEmpty.clearActions();
        props = {editable:true, ...propId, ...propBlockId, ...propDataText, ...propActiveElementId};
        component = renderComponentWithStore(CvElement,storeEmpty,props);
        let elem = component.root.find( el => el.type=='span');
        //клик по текстовому элементу
        elem.props.onClick();
        const expectedActions = [
            cvElement_activate(propDataText.data.style,propId.id),
        ];
        expect(storeEmpty.getActions()).toEqual(expectedActions);
    });

    it('Change text', () => {
        storeEmpty.clearActions();
        props = {editable:true, ...propId, ...propBlockId, ...propDataText, ...propActiveElementId};
        component = renderComponentWithStore(CvElement,storeEmpty,props);
        let elem = component.root.find( el => el.type=='span');
        //изменение текста в элементе
        const evt = {"target":{"innerText":'test'}};
        elem.props.onBlur(evt);
        const expectedActions = [
            cvElement_textUpdate(propBlockId.blockId,evt.target.innerText),
        ];
        expect(storeEmpty.getActions()).toEqual(expectedActions);
    });
});
