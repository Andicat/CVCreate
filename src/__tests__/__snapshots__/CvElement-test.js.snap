// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvElement render Рендер активного элемента 1`] = `
<span
  className="cv__element cv__element--text cv__element--active"
  contentEditable={false}
  data-elem={true}
  onBlur={[Function]}
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#8e9fa0",
      "color": "#000000",
      "fontFamily": "Roboto",
      "fontSize": "14px",
      "fontStyle": "normal",
      "fontWeight": "normal",
      "padding": "0px 0px 0px 1px",
      "textAlign": "start",
      "textDecoration": "none",
      "textTransform": "none",
    }
  }
  suppressContentEditableWarning={false}
>
  Text with background
</span>
`;

exports[`CvElement render Рендер без данных 1`] = `null`;

exports[`CvElement render Рендер блока прогресса 1`] = `
<div
  className="cv__element cv__element--progress"
  data-elem={true}
  onClick={[Function]}
  style={
    Object {
      "addcolor": "#E6E6E6",
      "maincolor": "#E05B49",
      "progress": 50,
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#E05B49",
        "width": "50%",
      }
    }
  />
  <div
    style={
      Object {
        "backgroundColor": "#E6E6E6",
        "width": "50%",
      }
    }
  />
</div>
`;

exports[`CvElement render Рендер блока с кружочками 1`] = `
<div
  className="cv__element cv__element--dots-row"
  data-elem={true}
  onClick={[Function]}
  style={
    Object {
      "addcolor": "#E6E6E6",
      "addcount": 3,
      "maincolor": "#E05B49",
      "maincount": 5,
      "radius": 10,
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#E05B49",
        "borderRadius": "50%",
        "height": "10px",
        "margin": "0 5px",
        "width": "10px",
      }
    }
  />
  <div
    style={
      Object {
        "backgroundColor": "#E05B49",
        "borderRadius": "50%",
        "height": "10px",
        "margin": "0 5px",
        "width": "10px",
      }
    }
  />
  <div
    style={
      Object {
        "backgroundColor": "#E05B49",
        "borderRadius": "50%",
        "height": "10px",
        "margin": "0 5px",
        "width": "10px",
      }
    }
  />
  <div
    style={
      Object {
        "backgroundColor": "#E05B49",
        "borderRadius": "50%",
        "height": "10px",
        "margin": "0 5px",
        "width": "10px",
      }
    }
  />
  <div
    style={
      Object {
        "backgroundColor": "#E05B49",
        "borderRadius": "50%",
        "height": "10px",
        "margin": "0 5px",
        "width": "10px",
      }
    }
  />
  <div
    style={
      Object {
        "backgroundColor": "#E6E6E6",
        "borderRadius": "50%",
        "height": "10px",
        "margin": "0 5px",
        "width": "10px",
      }
    }
  />
  <div
    style={
      Object {
        "backgroundColor": "#E6E6E6",
        "borderRadius": "50%",
        "height": "10px",
        "margin": "0 5px",
        "width": "10px",
      }
    }
  />
  <div
    style={
      Object {
        "backgroundColor": "#E6E6E6",
        "borderRadius": "50%",
        "height": "10px",
        "margin": "0 5px",
        "width": "10px",
      }
    }
  />
</div>
`;

exports[`CvElement render Рендер картинки 1`] = `
<img
  alt=""
  className="cv__element cv__element--image"
  data-elem={true}
  onClick={[Function]}
  src={null}
  style={
    Object {
      "borderColor": "#E05B49",
      "borderRadius": "50%",
      "borderStyle": "solid",
      "borderWidth": "3px",
      "file": "",
      "opacity": 1,
    }
  }
/>
`;

exports[`CvElement render Рендер текст 1`] = `
<span
  className="cv__element cv__element--text"
  contentEditable={false}
  data-elem={true}
  onBlur={[Function]}
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#8e9fa0",
      "color": "#000000",
      "fontFamily": "Roboto",
      "fontSize": "14px",
      "fontStyle": "normal",
      "fontWeight": "normal",
      "padding": "0px 0px 0px 1px",
      "textAlign": "start",
      "textDecoration": "none",
      "textTransform": "none",
    }
  }
  suppressContentEditableWarning={false}
>
  Text with background
</span>
`;

exports[`CvElement render Рендер текста с возможностью правки 1`] = `
<span
  className="cv__element cv__element--text"
  contentEditable={true}
  data-elem={true}
  onBlur={[Function]}
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#8e9fa0",
      "color": "#000000",
      "fontFamily": "Roboto",
      "fontSize": "14px",
      "fontStyle": "normal",
      "fontWeight": "normal",
      "padding": "0px 0px 0px 1px",
      "textAlign": "start",
      "textDecoration": "none",
      "textTransform": "none",
    }
  }
  suppressContentEditableWarning={true}
>
  Text with background
</span>
`;

exports[`CvElement render Рендер текстового блока 1`] = `
<div
  className="cv__element cv__element--group"
  style={Object {}}
>
  <span
    className="cv__element cv__element--text"
    contentEditable={false}
    data-elem={true}
    onBlur={[Function]}
    onClick={[Function]}
    style={
      Object {
        "color": "#000000",
        "fontFamily": "Roboto",
        "fontSize": "18px",
        "fontStyle": "normal",
        "fontWeight": "bold",
        "padding": "0px 0px 0px 1px",
        "textAlign": "start",
        "textDecoration": "none",
        "textTransform": "none",
      }
    }
    suppressContentEditableWarning={false}
  >
    Your position
  </span>
  <span
    className="cv__element cv__element--text"
    contentEditable={false}
    data-elem={true}
    onBlur={[Function]}
    onClick={[Function]}
    style={
      Object {
        "color": "#000000",
        "fontFamily": "Roboto",
        "fontSize": "18px",
        "fontStyle": "normal",
        "fontWeight": "normal",
        "padding": "0px 0px 0px 1px",
        "textAlign": "start",
        "textDecoration": "none",
        "textTransform": "none",
      }
    }
    suppressContentEditableWarning={false}
  >
    Company
  </span>
  <span
    className="cv__element cv__element--text"
    contentEditable={false}
    data-elem={true}
    onBlur={[Function]}
    onClick={[Function]}
    style={
      Object {
        "color": "#000000",
        "fontFamily": "Roboto",
        "fontSize": "16px",
        "fontStyle": "italic",
        "fontWeight": "normal",
        "padding": "0px 0px 0px 1px",
        "textAlign": "start",
        "textDecoration": "none",
        "textTransform": "none",
      }
    }
    suppressContentEditableWarning={false}
  >
    period
  </span>
  <span
    className="cv__element cv__element--text"
    contentEditable={false}
    data-elem={true}
    onBlur={[Function]}
    onClick={[Function]}
    style={
      Object {
        "color": "#000000",
        "fontFamily": "Roboto",
        "fontSize": "16px",
        "fontStyle": "normal",
        "fontWeight": "normal",
        "padding": "0px 0px 0px 1px",
        "textAlign": "start",
        "textDecoration": "none",
        "textTransform": "none",
      }
    }
    suppressContentEditableWarning={false}
  >
    your competencies and results
  </span>
</div>
`;

exports[`CvElement render Рендер цветного блока 1`] = `
<div
  className="cv__element cv__element--figure"
  data-elem={true}
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#6AABB5",
      "borderRadius": "50%",
      "opacity": 1,
    }
  }
/>
`;
